calico.go

```diff
@@ -312,6 +318,33 @@ func cmdAdd(args *skel.CmdArgs) error {
 	return types.PrintResult(result, cniVersion)
 }
 
+// sanitizeMesosLabel converts a string from a valid mesos label to a valid Calico label.
+// Mesos labels have no restriction outside of being unicode.
+func sanitizeMesosLabel(s string) string {
+	// Inspired by:
+	// https://github.com/projectcalico/libcalico-go/blob/2ff29bed865c4b364d4fcf1ad214b2bd8d9b4afa/lib/upgrade/converters/names.go#L39-L58
+	var (
```

Comment:
A bit weird to have a `var` section inside a function - could you just use `:=` here instead?

You might want to move this into one of the utility files as well.